crates/datasources/src/object_store/http.rs

```diff
@@ -326,10 +334,19 @@ impl ObjectStore for SimpleHttpStore {
             if content_length != 0 {
                 self.obj_store.get_opts(location, options).await
             } else {
-                self.simple_get_req(location.clone()).await.map_err(|e| {
+                self.simple_get_req(location.clone()).await.map_err(|err| {
+                    let err = if self.url.as_str().contains('*') {
+                        ObjectStoreSourceError::String(format!(
+                            "{err}\n\
+                            Note that globbing is not supported for HTTP.",
```

Comment:
I would avoid newlines in errors.

Reply:
The error is usually very long. Moreover, we just want to append a "note" here for the user (easier to read).

Reply:
I mean, I think it's fine.

If it's very long it's going to wrap anyway and adding our own new line character isn't going to actually be a great place in all cases.

My concern is that it's going to end up in log messages with a new line that's going to be hard to read and maybe hard to parse. 

I'd also change it to put the summary at the beginning: `HTTP sources do not support globbing: {err}`